sonar/modules/documents/dnb.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+#
+# Swiss Open Access Repository
+# Copyright (C) 2022 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Dnb API."""
+
+import base64
+import json
+
+import requests
+from flask import current_app
+
+
+class Dnb:
+    """Dnb class."""    
+
+    @classmethod
+    def headers(cls):
+        """."""
+        username = current_app.config.get('SONAR_APP_URN_DNB_USERNAME')
+        password = current_app.config.get('SONAR_APP_URN_DNB_PASSWORD')
+        string = f"{username}:{password}"
+        auth = base64.b64encode(string.encode()).decode('ascii')
+        return {
+            'Authorization': f'Basic {auth}',
+            'Content-Type': 'application/json',
+            'Accept': 'application/json'
+        }
+
+    @classmethod
+    def verify(cls, urn_code):
+        """Verify the existence for a URN.
+
+        :param urn_code: the urn code.
+        :returns: True if urn is registered, otherwise False.
+        """
+        # Documentation: https://wiki.dnb.de/display/URNSERVDOK/URN-Service+API
+        # https://wiki.dnb.de/display/URNSERVDOK/Beispiele%3A+URN-Verwaltung
+        answer = False
+        try:
+            response = requests.request(
+                'HEAD',
+                f"{current_app.config.get('SONAR_APP_URN_DNB_BASE_URL')}"\
+                f"/urns/urn/{urn_code}",
+                headers=cls.headers()
+            )
+            answer = response.status_code == 200
+        except Exception as error:
+            pass
```

Comment:
Should we log the error? Otherwise use: https://docs.sourcery.ai/refactorings/use-contextlib-suppress/